services/workers/src/test/java/org/sagebionetworks/table/worker/TableWorkerIntegrationTest.java

```diff
@@ -2386,6 +2384,36 @@ public void testNotHasQuery() throws Exception{
 		});
 	}
 
+	@Test
+	public void testCurrentUserQuery() throws Exception{
+		// setup an EntityId column.
+		ColumnModel userIDColumn = new ColumnModel();
+		userIDColumn.setColumnType(ColumnType.USERID);
+		userIDColumn.setName("userIDcolumn");
+		userIDColumn = columnManager.createColumnModel(adminUserInfo, userIDColumn);
+		schema = Lists.newArrayList(userIDColumn);
+		// build a table with this column.
+		createTableWithSchema();
+		TableStatus status = waitForTableProcessing(tableId);
+		assertTrue(TableState.AVAILABLE.equals(status.getState()));
+
+		RowSet rowSet = new RowSet();
+		rowSet.setRows(Lists.newArrayList(
+				TableModelTestUtils.createRow(null, null, adminUserInfo.getId().toString()),
+				TableModelTestUtils.createRow(null, null, "2"),
+				TableModelTestUtils.createRow(null, null, "3")));
+		rowSet.setHeaders(TableModelUtils.getSelectColumns(schema));
+		rowSet.setTableId(tableId);
+		referenceSet = appendRows(adminUserInfo, tableId,
+				rowSet, mockProgressCallback);
+
+		//query the column and check if the results match the userId
+		waitForConsistentQuery(adminUserInfo, "select * from " + tableId + " where userIDcolumn = CURRENT_USER()", null, null, (queryResult) -> {
+			assertEquals(1, queryResult.getQueryResults().getRows().size());

```

Comment:
Add assert on the number of values, and to make it consistent add the userId as the select clause